### PR TITLE
RepresentableStoreT

### PR DESCRIPTION
**Pull request by TimWSpence**
Add the `RepresentableStoreT` (co)monad-transformer. Unfortunately I don't _think_ it's possible to unify the existing `RepresentableStore` with this (by setting `W = Id`) due to bin-compat problems. If that is not the case then please do let me know!

### CODE REVIEW

File: core/src/main/scala/cats/data/RepresentableStore.scala [hunk 1]
```diff
@@ -3,7 +3,7 @@ package cats.data
 import cats.{Comonad, Functor, Representable}
 
 /**
- * A specialization of the `Store` comonad, for any `Representable` functor.
+ * A generalization of `StoreT`, where the underlying functor `F` hasfor any `Representable` functor.
```
Comment (ritschwumm) [inline core/src/main/scala/cats/data/RepresentableStore.scala:6]: "hasfor" ?

**Review by larsrh — APPROVED** [review]

**Comment by TimWSpence** [comment]
@larsrh thanks for reviewing! Is this just waiting for someone else to look at it or do I need to do anything else?

**Comment by larsrh** [comment]
I was hoping to get another :+1: but I think this addition is good regardless :smile:

**Comment by TimWSpence** [comment]
> I was hoping to get another 👍 but I think this addition is good regardless 😄

😂  Thanks!!

**Pull request merged**